### PR TITLE
Expose configuration for quartz scheduler performance tuning

### PR DESCRIPTION
**Pull request by fikrutm**
Allows to configure maxBatchSize and batchTimeWindow to leverage the  batch mode as described in [#29319](https://github.com/quarkusio/quarkus/issues/29319) .
@machi1990

### CODE REVIEW

**Comment by machi1990** [comment]
Thanks @fikrutm 

Can you sort out the commits and remove the second merge commit? Thanks.

File: extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzRuntimeConfig.java
Comment (machi1990) [inline extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzRuntimeConfig.java:22]: Let's also allow configuring `acquireTriggersWithinLock`. This config item has to default to `true`? 

This property has to be if JDBC JobStore is used i.e anywhere here https://github.com/quarkusio/quarkus/blob/19bded9c48d6d65fb4615b46e1d80d8f74930d57/extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzSchedulerImpl.java#L577-L587

**Review by machi1990 — CHANGES_REQUESTED** [review]
see the comments

File: extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzRuntimeConfig.java
Comment (fikrutm) [reply extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzRuntimeConfig.java:22]: Yes the acquireTriggersWithinLock has to be true if JDBC JobStore is used, but the default value is either false or true. As of Quartz 2.1.1 “true is the default if batchTriggerAquisitionMaxCount is set to >1 otherwise false by defualt.

File: extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzSchedulerImpl.java
Comment (machi1990) [inline extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzSchedulerImpl.java:591]: I think we can remove this line (https://github.com/quarkusio/quarkus/blob/88bfa7eccef25b99a47f510ebb99ea679fe48d1a/extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzSchedulerImpl.java#L591), since it is done above https://github.com/quarkusio/quarkus/blob/88bfa7eccef25b99a47f510ebb99ea679fe48d1a/extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzSchedulerImpl.java#L588

**Review by machi1990 — CHANGES_REQUESTED** [review]
Added one comment

Can you also squash the commit before we merge this one? Thanks

File: extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzSchedulerImpl.java
Comment (fikrutm) [reply extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzSchedulerImpl.java]: Done.

**Review by machi1990 — APPROVED** [review]
Thanks @fikrutm

**Comment by machi1990** [comment]
@mkouba can you've a look as well?

**Review by mkouba — APPROVED** [review]
Looks good. I assume that there is no easy way to test those config properties, right?

File: extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzRuntimeConfig.java
Comment (geoand) [inline extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzRuntimeConfig.java:26]: Wouldnt't it be better for users and also more consistent with the rest of the properties if this where a `Duration`?

Comment (mkouba) [reply extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzRuntimeConfig.java:26]: It depends. I think that it's kind of reasonable to follow the way Quartz defines the property, i.e. "the amount of time in milliseconds" so that users can just copy and paste properties from the quartz config file. OTOH `Duration` is indeed an idiomatic way in Quarkus.

Comment (geoand) [reply extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzRuntimeConfig.java:26]: `Duration` is also used [here](https://github.com/quarkusio/quarkus/blob/2.14.2.Final/extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzRuntimeConfig.java#L59).

Comment (mkouba) [reply extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzRuntimeConfig.java:26]: > `Duration` is also used [here](https://github.com/quarkusio/quarkus/blob/2.14.2.Final/extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzRuntimeConfig.java#L59).

Yes, but this config property is not passed directly to quartz. It's a `quarkus-quartz` -specific property.

Comment (geoand) [reply extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzRuntimeConfig.java:26]: Sure, but it creates a suboptimal user experience to not be consistent

Comment (machi1990) [reply extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzRuntimeConfig.java:26]: > Sure, but it creates a suboptimal user experience to not be consistent

To be fair, I think someone wanting to configure this will be pretty much experienced with quartz so they may want to just use the config value they are already used to.

Comment (geoand) [reply extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzRuntimeConfig.java:26]: Yeah, I understand. I'll leave this up to you guyes

Comment (mkouba) [reply extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzRuntimeConfig.java:26]: I'm not sure. This config property will be very likely set by advanced users who seek consistency with Quartz. Just my 2c. That said, I don't have a problem with `Duration`.

**Pull request merged**
